src/google/protobuf/util/internal/proto_writer.h

```diff
@@ -76,7 +76,8 @@ class LIBPROTOBUF_EXPORT ProtoWriter : public StructuredObjectWriter {
  public:
 // Constructor. Does not take ownership of any parameter passed in.
   ProtoWriter(TypeResolver* type_resolver, const google::protobuf::Type& type,
-              strings::ByteSink* output, ErrorListener* listener);
+              strings::ByteSink* output, ErrorListener* listener,
+              bool ignore_unknown_fields = false);
```

Comment:
Use a setter instead? We may add more flags in the future and it's not very nice to put all flags in the constructor.